src/js/plugin/PluginLoader.js

```diff
@@ -0,0 +1,98 @@
+import config from "../config/config";
+
+import URLUtil from "../helpers/URLUtil";
+import PluginDispatcher from "./external/PluginDispatcher";
+import PluginEvents from "./external/PluginEvents";
+import PluginMountPoints from "./external/PluginMountPoints";
+import PluginActions from "./external/PluginActions";
+import PluginDispatcherProxy from "./PluginDispatcherProxy";
+
+const PLUGIN_STARTUP_TIMEOUT = 10000; // in ms
+
+const PluginLoader = {
+  load: function (pluginId, pluginURI) {
+    return new Promise(function (resolve, reject) {
+
+      // Create plugin host (hidden iframe)
+      const pluginHost = document.createElement("iframe");
+      pluginHost.setAttribute("scrolling","no");
+      pluginHost.setAttribute("width","1");
+      pluginHost.setAttribute("height","1");
+      pluginHost.setAttribute("frameborder","0");
+      pluginHost.setAttribute("style",
+        "position:absolute; top:-1px; left:-1px;");
+      pluginHost.setAttribute("sandbox","allow-same-origin allow-scripts");
+      pluginHost.setAttribute("src", URL.createObjectURL(new Blob([
+        `<html><head></head><body>${pluginId}</body></html>`
+      ], {type: "text/html"})));
+
+      function handlePluginHostLoad() {
+        pluginHost.removeEventListener("load", handlePluginHostLoad);
+        pluginHost.removeEventListener("error", handlePluginHostError);
+
+        const pluginWindow = pluginHost.contentWindow;
+        const pluginDocument = pluginWindow.document;
+        var pluginTimeout = null;
+
+        // Inject plugin interface
+        pluginWindow.marathonPluginInterface = Object.freeze({
```

Comment:
Hm. I've removed the file as there is a custom PluginDispatcherProxy for every plugin. We could create an MarathonPluginInterface "class" and create an instance within the loader, if you like that better.

I don't feel that we lost anything with the renaming, it could only be a *UI* plugin interface and not . I renamed it mostly because it's a property of the plugin window providing the plugin interface and not a plugin **application** interface. Why have you named it that way? Do you want me to rename it? 

BTW: no one should ever touch the interface directly, but use the sdk instead.